### PR TITLE
fix: Auto creation of Stock Entry on Submit of Pick List

### PR DESCRIPTION
**Pull request by bhattdevarsh**
[TASK](https://app.asana.com/0/1192407897953440/1199522618328717/f)

stock entry_ on pick list

### CODE REVIEW

File: erpnext/stock/doctype/pick_list/pick_list.py [hunk 1]
```diff
@@ -580,11 +577,16 @@ def update_stock_entry_based_on_material_request(pick_list, stock_entry):
...
-	for location in pick_list.locations:
-		item = frappe._dict()
-		update_common_item_properties(item, location)
-
+def update_stock_entry_items_with_no_reference(location, stock_entry):
+	item = frappe._dict()
+	update_common_item_properties(item, location)
+	if location.source_package_tag:
+		item.package_tag = location.source_package_tag
+		# item.s_warehouse = location.warehouse
```
Comment (bhattdevarsh) [inline erpnext/stock/doctype/pick_list/pick_list.py]: @neilLasrado Currently we cannot have target warehouse and source warehouse as same. so which warehouse going to get selected as a target while `Repack` is made?  also `Pick List` is created for `Delivery`. so technically the packages are not relocated from one warehouse to other. it is going out to the Customer. so making entry in Target warehouse has no meaning. please explain how can we deal with this.

File: erpnext/stock/doctype/pick_list/pick_list.py [hunk 2]
```diff
@@ -214,6 +215,28 @@ def set_per_picked(self):
 
 			frappe.db.set_value("Sales Order", self.locations[0].sales_order, "per_picked", per_picked, update_modified=False)
 
+	def create_stock_entry(self):
+		validate_item_locations(self)
+		for location in self.locations:
+			if stock_entry_exists(self.get('name')):
+				return frappe.msgprint(_('Stock Entry has been already created against this Pick List'))
+			stock_entry = frappe.new_doc('Stock Entry')
+			stock_entry.self = self.get('name')
```
Comment (hrwX) [inline erpnext/stock/doctype/pick_list/pick_list.py]: Why are you adding a `self` attr to `stock_entry` object ?

File: erpnext/stock/doctype/pick_list/pick_list.py [hunk 3]
```diff
@@ -214,6 +215,28 @@ def set_per_picked(self):
...
+			purpose = self.get('purpose')
+			if purpose in "Delivery":
```
Comment (hrwX) [inline erpnext/stock/doctype/pick_list/pick_list.py]:
```suggestion
			if self.get('purpose') == "Delivery":
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/pick_list/pick_list.py (hunk 1 above)
Comment (neilLasrado) [reply erpnext/stock/doctype/pick_list/pick_list.py]: @bhattdevarsh is this still an issue?

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**